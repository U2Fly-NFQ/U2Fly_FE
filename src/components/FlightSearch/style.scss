@use '../../assets/scss/abstracts' as var;

.flightSearch {
  padding: 30px;
  background: #ffffff;
  box-shadow: 4px 14px 28px rgb(0 0 0 / 10%);
  border-radius: 20px;
  position: relative;

  &Passenger {
    min-height: 32px;
    font-size: 18px;
    font-weight: 600;
  }

  &Categories {
    margin-bottom: 20px;
  }

  &Box {
    background: var.$bg-search-box;
    padding: 10px 10px 10px 20px;
    border-radius: 10px;
    position: relative;
    height: 100%;

    .ant-select.ant-select-in-form-item {
      width: 70%;
    }

    & div {
      margin-bottom: 0 !important;
    }

    &__Icon {
      position: absolute;
      top: 20%;
      right: 12%;
      font-size: 26px;
    }
  }

  &Input {
    height: 35px;
    padding: 0;
    width: 70%;
    font-size: 18px;

    & .ant-select-selector {
      background: transparent !important;
      border: none !important;
      outline: none !important;
      padding: 0 !important;

      & .ant-select-selection-search {
        left: 0;
        right: 0;
      }

      & input {
        color: var.$black-color !important;
        font-weight: 600 !important;
      }
    }
  }

  &Label {
    display: block;
    color: var.$paragraph-color !important;
    line-height: 28px;
    color: var(--paragraph-color);
    font-weight: 400;
    font-family: 'Poppins', sans-serif;
    margin-bottom: 0;
    font-size: 14px;
  }

  &Selected {
    display: block;
    font-size: 12px;
    padding-top: 5px;
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
    margin: 0;
  }

  & .formControl {
    padding-top: 30px;
    display: flex;
    justify-content: center;
  }
}
