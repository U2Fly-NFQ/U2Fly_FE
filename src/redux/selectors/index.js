export * from './auth'
export * from './bookingFlightSelector'
export * from './ticket'
