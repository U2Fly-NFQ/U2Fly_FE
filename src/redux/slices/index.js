export * from './searchSlice'
export * from './filterSlice'
export * from './flightSlice'
