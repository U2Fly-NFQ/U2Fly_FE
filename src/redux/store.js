import { configureStore } from '@reduxjs/toolkit'
import bookingFlightsSlice from './slices/bookingFlightsSlice'

// import loginSlice from './slices/loginSlice'
import flightSlice from './slices/flightSlice'
import airportSlice from './slices/airportSlice'

const store = configureStore({
  reducer: {
    // login: loginSlice.reducer,
    bookingFlight: bookingFlightsSlice.reducer,
    flights: flightSlice.reducer,
    airports: airportSlice.reducer,
  },
})

export default store
