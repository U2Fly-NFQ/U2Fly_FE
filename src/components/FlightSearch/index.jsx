import React, { useEffect, useState } from 'react'
import {
  useNavigate,
  useSearchParams,
  createSearchParams,
} from 'react-router-dom'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import { motion, AnimatePresence } from 'framer-motion'

import './style.scss'
import {
  Col,
  Row,
  DatePicker,
  Radio,
  Popover,
  InputNumber,
  Select,
  Button,
  Tooltip,
  Modal,
  Space,
} from 'antd'

import moment from 'moment'
import { fetchAirports } from '../../redux/slices/airportSlice'
import { updateLs, getLsObj } from '../../utils/localStorage'
import { CloseOutlined } from '@ant-design/icons'

const { Option } = Select

export const variantsSelector = {
  show: {
    opacity: 1,
    y: 0,
    transition: {
      ease: 'easeOut',
      duration: 0.5,
    },
  },
  hide: {
    y: -20,
    opacity: 0,
  },
}

export default function FlightSearch() {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const [searchParams, setSearchParams] = useSearchParams()

  const airports = useSelector((state) => state.airports.data)
  const [searchTo, setSearchTo] = useState('')
  const [searchFrom, setSearchFrom] = useState('')

  const [from, setFrom] = useState(undefined)
  const [to, setTo] = useState(undefined)
  const [ticketType, setTicketType] = useState('oneWay')
  const [journeyDay, setJourneyDay] = useState(moment())
  const [passengerClass, setPassengerClass] = useState('economy')
  const [returnDate, setReturnDate] = useState(moment().add(3, 'days'))
  const [passengerNumber, setPassengerNumber] = useState(1)
  const [modalContent, setModalContent] = useState('')

  const flightStorage = getLsObj('flight')

  useEffect(() => {
    dispatch(fetchAirports())
  }, [])

  useEffect(() => {
    const {
      seatType,
      seatAvailable,
      departure,
      arrival,
      startDate,
      ticketType,
      startDateRoundTrip,
    } = flightStorage

    // If exist query in local storage
    if (departure) setFrom(departure)
    if (arrival) setTo(arrival)
    // if (startDate) setJourneyDay(moment(startDate))
    if (seatType) setPassengerClass(seatType)
    if (seatAvailable) setPassengerNumber(seatAvailable)
    if (ticketType) setTicketType(ticketType)
    // if (startDateRoundTrip) setReturnDate(moment(returnDate))

    // If exist query on URL
    const departureParam = searchParams.get('departure')
    const arrivalParam = searchParams.get('arrival')
    const startDateParam = searchParams.get('startDate')
    const seatTypeParam = searchParams.get('seatType')
    const seatAvailableParam = searchParams.get('seatAvailable')
    const ticketTypeParam = searchParams.get('ticketType')
    const startDateRoundTripParam = searchParams.get('startDateRoundTrip')

    if (departureParam) setFrom(departureParam)
    if (arrivalParam) setTo(arrivalParam)
    if (startDateParam) setJourneyDay(moment(startDateParam))
    if (seatTypeParam) setPassengerClass(seatTypeParam)
    if (seatAvailableParam) setPassengerNumber(seatAvailableParam)
    if (ticketTypeParam) setTicketType(ticketTypeParam)
    if (startDateRoundTripParam) setReturnDate(moment(startDateRoundTripParam))
  }, [searchParams])

  const onFinish = async () => {
    if (!from || !to) {
      setModalContent(
        'Please enter the origin, destination and your travel date to proceed'
      )
      return
    }

    const searchQuery = {
      departure: from,
      arrival: to,
      startDate: journeyDay.format('YYYY-MM-DD'),
      seatType: passengerClass,
      seatAvailable: passengerNumber,
      ticketType,
      startDateRoundTrip:
        ticketType === 'roundTrip' ? returnDate.format('YYYY-MM-DD') : '',
    }

    updateLs('flight', searchQuery)

    navigate({
      pathname: '/flights',
      search: createSearchParams(searchQuery).toString(),
    })
  }

  const onChangeTicketType = (value) => {
    setTicketType(value)
  }

  const onChangeFrom = (option) => {
    if (option) setFrom(option)
    setSearchFrom('')
  }

  const onChangeTo = (option) => {
    if (option) setTo(option)
    setSearchTo('')
  }

  const clearFrom = () => {
    setFrom('')
  }

  const clearTo = () => {
    setTo('')
  }

  const passengerPopover = (
    <>
      <label className="flightSearchLabel">
        {t('search_form.No. of Passengers')}
      </label>
      <InputNumber
        min={1}
        max={5}
        value={passengerNumber}
        onChange={(value) => {
          setPassengerNumber(value)
        }}
      />
      <label
        className="flightSearchLabel"
        style={{
          marginTop: '10px',
        }}
      >
        {t('search_form.class')}
      </label>
      <Radio.Group
        buttonStyle="outlined"
        value={passengerClass}
        onChange={(e) => setPassengerClass(e.target.value)}
      >
        <Radio.Button value="economy">{t('search_form.economy')}</Radio.Button>
        <Radio.Button value="business">
          {t('search_form.business')}
        </Radio.Button>
      </Radio.Group>
    </>
  )

  const [showAskModal, setShowAskModal] = useState(true)

  const chooseDeparture = (value) => {
    updateLs('flight', {
      departure: value,
    })
    setFrom(value)
  }

  return (
    <div className="flightSearchContainer">
      {!flightStorage.departure && showAskModal && (
        <Modal
          visible={true}
          style={{
            borderRadius: '12px',
            overflow: 'hidden',
            paddingBottom: 0,
          }}
          maskStyle={{
            backgroundColor: 'rgba(14, 134, 212, 0.4)',
          }}
          footer={false}
          onCancel={() => setShowAskModal(false)}
        >
          Hey buddy, where is your departure?
          <Select
            size="large"
            clearIcon={<CloseOutlined />}
            autoFocus
            showSearch
            value={airports[0]}
            showArrow={false}
            onSelect={chooseDeparture}
            bordered={false}
            defaultOpen
            placement="bottomLeft"
            style={{
              width: '100%',
              borderBottom: '1px solid #ddd',
            }}
            dropdownStyle={{
              borderRadius: '10px',
            }}
            placeholder="Flying from..."
          >
            {airports.map((airport) => {
              return (
                <Option key={airport.iata} value={airport.iata}>
                  {airport.city} ({airport.iata})
                </Option>
              )
            })}
          </Select>
        </Modal>
      )}
      <Modal
        visible={modalContent && true}
        onOk={() => setModalContent('')}
        onCancel={() => setModalContent('')}
        centered
        cancelButtonProps={{
          hidden: 'true',
        }}
      >
        {modalContent}
      </Modal>
      <div className="flightSearch">
        <div className="flightSearchCategories">
          <Radio.Group
            value={ticketType}
            onChange={(e) => onChangeTicketType(e.target.value)}
            buttonStyle="solid"
          >
            <Radio.Button value="oneWay">
              {t('search_form.one_way')}
            </Radio.Button>
            <Radio.Button value="roundTrip">
              {t('search_form.round_trip')}
            </Radio.Button>
          </Radio.Group>
        </div>

        <Row gutter={[24, 24]} className="flightSearchBody">
          <Col span={24} md={12} lg={6}>
            <div className="flightSearchBox">
              <i className="flightSearchBox__Icon fa-solid fa-plane-departure"></i>
              <label className="flightSearchLabel">
                {t('search_form.from')}
              </label>
              <motion.div
                key={from}
                variants={variantsSelector}
                animate={'show'}
                initial="hide"
              >
                <Select
                  size="large"
                  allowClear
                  clearIcon={<CloseOutlined />}
                  onClear={clearFrom}
                  showSearch
                  value={from}
                  showArrow={false}
                  filterOption={() => true}
                  onChange={onChangeFrom}
                  onSearch={(text) => setSearchFrom(text)}
                  bordered={false}
                  style={{
                    width: '100%',
                    borderBottom: '1px solid #ddd',
                  }}
                  dropdownStyle={{
                    borderRadius: '10px',
                  }}
                  placeholder="Flying from..."
                >
                  {airports.map((airport) => {
                    return (
                      airport.city
                        .toLowerCase()
                        .includes(searchFrom.toLowerCase()) &&
                      airport.iata !== to && (
                        <Option key={airport.iata} value={airport.iata}>
                          {airport.city} ({airport.iata})
                        </Option>
                      )
                    )
                  })}
                </Select>
              </motion.div>
            </div>
          </Col>
          <Col span={24} md={12} lg={6}>
            <div className="flightSearchBox">
              <i className="flightSearchBox__Icon fa-solid fa-plane-arrival"></i>
              <label className="flightSearchLabel">{t('search_form.to')}</label>
              <motion.div
                key={to}
                variants={variantsSelector}
                animate={'show'}
                initial="hide"
              >
                <Select
                  size="large"
                  showSearch
                  allowClear
                  clearIcon={<CloseOutlined />}
                  onClear={clearTo}
                  value={to}
                  showArrow={false}
                  filterOption={() => true}
                  onChange={onChangeTo}
                  onSearch={(text) => setSearchTo(text)}
                  bordered={false}
                  style={{
                    width: '100%',
                    borderBottom: '1px solid #ddd',
                  }}
                  dropdownStyle={{
                    borderRadius: '10px',
                  }}
                  placeholder="Flying to..."
                >
                  {airports.map((airport) => {
                    return (
                      airport.city
                        .toLowerCase()
                        .includes(searchTo.toLowerCase()) &&
                      airport.iata !== from && (
                        <Option key={airport.iata} value={airport.iata}>
                          {airport.city} ({airport.iata})
                        </Option>
                      )
                    )
                  })}
                </Select>
              </motion.div>
            </div>
          </Col>
          <Col span={24} md={12} lg={7}>
            <div className="flightSearchBox">
              <Row gutter={[8, 8]} justify="center">
                <Col sm={12} xs={24}>
                  <label className="flightSearchLabel">
                    {t('search_form.journey_date')}
                  </label>
                  <DatePicker
                    className="journeyDate"
                    allowClear={false}
                    disabledDate={(current) => {
                      return (
                        moment().add(-1, 'days') > current ||
                        moment().add(1, 'months') <= current
                      )
                    }}
                    value={journeyDay}
                    onChange={(date) => setJourneyDay(date)}
                    format={'MM/DD/YY'}
                  />
                </Col>
                <Col sm={12} xs={24}>
                  <label className="flightSearchLabel">
                    {t('search_form.return_date')}
                  </label>
                  <AnimatePresence>
                    {(ticketType === 'roundTrip' && (
                      <motion.div
                        initial={{ x: -100, opacity: 0 }}
                        animate={{
                          x: 0,
                          opacity: 1,
                          transition: { duration: 0.6 },
                        }}
                        exit={{
                          x: -100,
                          opacity: 0,
                          transition: { duration: 0.6 },
                        }}
                      >
                        <DatePicker
                          allowClear={false}
                          disabledDate={(current) => {
                            return (
                              journeyDay > current ||
                              moment().add(1, 'month') <= current
                            )
                          }}
                          value={returnDate}
                          onChange={(date) => setReturnDate(date)}
                          format={'MM/DD/YY'}
                        />
                      </motion.div>
                    )) || (
                      <Space
                        style={{
                          display: 'block',
                        }}
                      >
                        <Button
                          type="link"
                          onClick={() => setTicketType('roundTrip')}
                          style={{
                            paddingLeft: 0,
                          }}
                        >
                          {t('search_form.Add return date')}
                        </Button>
                      </Space>
                    )}
                  </AnimatePresence>
                </Col>
              </Row>
            </div>
          </Col>
          <Col span={24} md={12} lg={5}>
            <Popover
              content={passengerPopover}
              trigger="click"
              placement="bottomRight"
            >
              <div className="flightSearchBox">
                <label className="flightSearchLabel">
                  {t('search_form.passenger')} & {t('search_form.class')}
                </label>
                <div className="flightSearchPassenger">
                  {passengerNumber} {t('search_form.passenger')}
                </div>
                <p className="flightSearchSelected">
                  {t(`search_form.${passengerClass}`)}
                </p>
              </div>
            </Popover>
          </Col>
        </Row>
        <div className="formControl">
          <Tooltip className="search-submit">
            <button
              type="primary"
              size="large"
              className="searchBtn btn btn-md btn-primary"
              onClick={onFinish}
            >
              {t('cta.search')}{' '}
            </button>
          </Tooltip>
        </div>
      </div>
    </div>
  )
}
