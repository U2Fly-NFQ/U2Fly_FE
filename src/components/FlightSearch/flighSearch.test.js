import { I18nextProvider } from 'react-i18next'
import i18n from '../../translations'
import renderer from 'react-test-renderer'
import React, { Suspense } from 'react'
import { store } from '../../redux/store'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import FlightSearch from './'
import { ConfigProvider } from 'antd'

describe('Flight search', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Provider store={store}>
            <ConfigProvider>
              <Router>
                <Suspense>
                  <FlightSearch />
                </Suspense>
              </Router>
            </ConfigProvider>
          </Provider>
        </I18nextProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
