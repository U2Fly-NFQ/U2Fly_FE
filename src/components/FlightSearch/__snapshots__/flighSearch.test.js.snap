// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight search test Flight search component renders correctly 1`] = `
<div
  className="flightSearchContainer"
>
  <div
    className="flightSearch"
  >
    <div
      className="flightSearchCategories"
    >
      <div
        className="ant-radio-group ant-radio-group-outline"
        style={
          Object {
            "borderRadius": "12px",
            "padding": "10px",
          }
        }
      >
        <label
          className="ant-radio-wrapper ant-radio-wrapper-checked"
        >
          <span
            className="ant-radio ant-radio-checked"
            style={Object {}}
          >
            <input
              checked={true}
              className="ant-radio-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="oneWay"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
          <span>
            One Way
          </span>
        </label>
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              type="radio"
              value="roundTrip"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
          <span>
            Round Trip
          </span>
        </label>
      </div>
    </div>
    <div
      className="ant-row flightSearchBody"
      style={
        Object {
          "marginBottom": -12,
          "marginLeft": -12,
          "marginRight": -12,
          "marginTop": -12,
        }
      }
    >
      <div
        className="ant-col ant-col-24 ant-col-md-12 ant-col-lg-6"
        style={
          Object {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <div
          className="flightSearchBox"
        >
          <i
            className="flightSearchBox__Icon fa-solid fa-plane-departure"
          />
          <label
            className="flightSearchLabel"
          >
            From
          </label>
          <div
            className="ant-select ant-select-lg ant-select-borderless ant-select-single ant-select-allow-clear ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "borderBottom": "1px solid #ddd",
                "width": "100%",
              }
            }
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  disabled={false}
                  id="rc_select_TEST_OR_SSR"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              >
                Flying from...
              </span>
            </div>
          </div>
          <p
            className="flightSearchSelected"
          />
        </div>
      </div>
      <div
        className="ant-col ant-col-24 ant-col-md-12 ant-col-lg-6"
        style={
          Object {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <div
          className="flightSearchBox"
        >
          <i
            className="flightSearchBox__Icon fa-solid fa-plane-arrival"
          />
          <label
            className="flightSearchLabel"
          >
            To
          </label>
          <div
            className="ant-select ant-select-lg ant-select-borderless ant-select-single ant-select-allow-clear ant-select-show-search"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            style={
              Object {
                "borderBottom": "1px solid #ddd",
                "width": "100%",
              }
            }
          >
            <div
              className="ant-select-selector"
              onClick={[Function]}
              onMouseDown={[Function]}
            >
              <span
                className="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autoComplete="off"
                  className="ant-select-selection-search-input"
                  disabled={false}
                  id="rc_select_TEST_OR_SSR"
                  onChange={[Function]}
                  onCompositionEnd={[Function]}
                  onCompositionStart={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onPaste={[Function]}
                  readOnly={false}
                  role="combobox"
                  style={
                    Object {
                      "opacity": null,
                    }
                  }
                  type="search"
                  unselectable={null}
                  value=""
                />
              </span>
              <span
                className="ant-select-selection-placeholder"
              >
                Flying to...
              </span>
            </div>
          </div>
          <p
            className="flightSearchSelected"
          />
        </div>
      </div>
      <div
        className="ant-col ant-col-24 ant-col-md-12 ant-col-lg-7"
        style={
          Object {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <div
          className="flightSearchBox"
        >
          <div
            className="ant-row ant-row-center"
            style={
              Object {
                "marginBottom": -4,
                "marginLeft": -4,
                "marginRight": -4,
                "marginTop": -4,
              }
            }
          >
            <div
              className="ant-col ant-col-12"
              style={
                Object {
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                }
              }
            >
              <label
                className="flightSearchLabel"
              >
                Journey Date
              </label>
              <div
                className="ant-picker journeyDate"
                onMouseUp={[Function]}
              >
                <div
                  className="ant-picker-input"
                >
                  <input
                    autoComplete="off"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    placeholder="Select date"
                    readOnly={true}
                    size={12}
                    title="07/07/22"
                    value="07/07/22"
                  />
                  <span
                    className="ant-picker-suffix"
                  >
                    <span
                      aria-label="calendar"
                      className="anticon anticon-calendar"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="calendar"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
            <div
              className="ant-col ant-col-12"
              style={
                Object {
                  "paddingBottom": 4,
                  "paddingLeft": 4,
                  "paddingRight": 4,
                  "paddingTop": 4,
                }
              }
            >
              <label
                className="flightSearchLabel"
              >
                Return Date
              </label>
              <div
                className="ant-space ant-space-horizontal ant-space-align-center"
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <button
                    className="ant-btn ant-btn-link"
                    disabled={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "paddingLeft": 0,
                      }
                    }
                    type="button"
                  >
                    <span>
                      Add return date
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-col ant-col-24 ant-col-md-12 ant-col-lg-5"
        style={
          Object {
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <div
          className="flightSearchBox"
          onClick={[Function]}
          onMouseDown={[Function]}
          onTouchStart={[Function]}
        >
          <label
            className="flightSearchLabel"
          >
            Passenger
             & 
            Class
          </label>
          <div
            className="flightSearchPassenger"
          >
            1 passenger
          </div>
          <p
            className="flightSearchSelected"
          >
            Economy
          </p>
        </div>
      </div>
    </div>
    <div
      className="formControl"
    >
      <button
        className="searchBtn btn btn-md btn-primary search-submit"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="large"
        type="primary"
      >
        Search
         
        One Way
      </button>
    </div>
  </div>
</div>
`;
